Add .gitattributes to specify LF line endings in files

sophm256 opened the pull request.

elmq0022 commented:
@sophm256, I tested this this morning on a windows 10 home machine with docker toolbox and it works just fine for me. 

Please, remove this from draft status so we can merge it into master.  I feel this is "fixed enough" at this point as it's the development environment and not the site.  We now have a reasonable way for windows users to jump into the web-guild which is a *big win*.  If there are remaining issues, let's fix those as we go.

In regards to instructions, let's keep it very minimal, something along the lines of:

> For windows users not using windows 10 pro, we recommend using docker toolbox available at ... with the default virtualbox virtual machine's port 8000 forwarded to the local machine's port 8000"  
> 
> Note: A .gitattributes file has been provided to keep line endings as LF instead of CRLF on checkout and commit. Issues related to developing on windows with docker toolbox are likely related to an incorrect line ending. 

Good work on this one. This looks like a simple commit, but I know and appreciate all the effort that went into it.  :+1:

sophm256 commented:
@elmq0022 Appreciate all of your help on this one. It was a lot of work for a simple solution.

sophm256 commented on README.md:
```diff
@@ -1,22 +1,26 @@
 * docker-compose - https://docs.docker.com/compose/install/
 * make - https://www.gnu.org/software/make/
 
+For Windows users not using Windows 10 Pro, we recommend using Docker Toolbox from https://docs.docker.com/toolbox/toolbox_install_windows/ . After following the README instructions below, select the `default` virtual machine in VirtualBox. Set up port forwarding from the Guest Port as 8000 to the Host Port as 8000. Go to `localhost:8000` on your browser to see the site.
```
Sure, I can implement the changes. Would it be alright with you if instead of saying `Set up port forwarding from the Guest Port as 8000 to the Host Port as 8000.`,  I put `Set up NAT port forwarding, where the Guest Port is 8000 and the Host Port is 8000. Guest IP and Host IP are left empty.` Either case is fine with me, but I thought the 2nd version would be more clear.

elmq0022 replied:
The second version is good.  You can go with that.

sophm256 commented:
I made a few commits since I last commented. In my most recent commit, I've differentiated the links for Mac and Linux, Windows 10 Pro, and Windows 7 to Non Pro Users. I've also added a Chocolatey link. And I've change a few lines in the instructions. Nick's instructions should probably go in where I've written `Use the package manager Chocolatey to install Make.`

sophm256 commented:
@elmq0022  and @nick-symon I've incorporated the info that Nick wrote regarding Chocolatey and Make into the readme.

@nick-symon I've simplified some of the wording. I've also tried to word it in a similar style as the rest of the readme. I left out the last sentence because that sentence referred to the Docker Quickstart, which I think is only in the Docker Toolbox  and not in the Docker Desktop for Windows version. So I talk about Docker Quickstart in the "Windows 7 to 10 " section. 

@elmq0022 Please let me know if you'd like me to make any changes. Thanks!

nick-symon commented:
@sophm256 Looks great! Thanks for all of the hard work!

sophm256 commented:
@nick-symon Thanks for your help!
